### DISCUSSION TITLE
Can i debug an extension download from vscode markeplace?

### DISCUSSION THREAD

**Discussion by gziz** [microsoft/vscode-discussions#751] [Extension Development QnA] [answered]
I'm working on an extension. However, when publishing and downloading it from the vscode marketplace, I'm getting some errors, I'm not getting when debugging my developer version.

**Comment by HatemMn** [comment, marked as answer]
Vs code extensions are installed in 

- Windows %USERPROFILE%\.vscode\extensions
- macOS ~/.vscode/extensions
- Linux ~/.vscode/extensions

You can navigate there, open the folder of your extension with vscode and see if something is looking wrong. To debug from within vscode, create you [launch.config](https://code.visualstudio.com/api/get-started/your-first-extension#debugging-the-extension) file and start debugging like you would usually do.

You might also disable other installed extensions before doing so and then debug, as they might cause the issue.

**Reply by gziz** [reply]
It was related to me not having the vscode nor sqlite3 package installed in the second device.
I'm assuming bundling should solve the problem right?

**Reply by HatemMn** [reply]
I do not really know how your extension is bundled, but maybe you have set them as devDependencies ( and not actual dependencies ) in your package.json ?

**Comment by r3m0t** [comment]
You can download the vsix from the VS Code Marketplace website, unzip and compare to the vsix that you build yourself with the `vsce` command.

You can also use "Attach to Node Process" from another VS Code window to debug the extension host process.

**Comment by alefragnani** [comment]
Hi @gziz ,

Could you please provide an error log when your extension is being activated and/or executed? Without that, it would be difficult to identify what may be happening.

If this is not possible for any reason, you could also reproduce the error without publishing/downloading the extension, if you `package` the extension instead of `publish` (using the **vsce** tools) and install the created `.vsix` file on your VS Code. 

A `.vsix` can fail to work (even activate) for a variety of reasons, from missing dependencies, incompatible dependencies, incompatible platforms, edge case testing scenarios and so on. The error log should give a starting point to find out where is the problem.

Hope this helps

**Reply by gziz** [reply]
It was related to me not having the vscode nor sqlite3 package installed in the second device.
I'm assuming bundling should solve the problem right?
